controller/app/validators/key_content_validator.rb

```diff
@@ -0,0 +1,15 @@
+class KeyContentValidator < ActiveModel::Validator
+   def validate(record)
+     key = Net::SSH::KeyFactory.load_data_public_key(record.type + " " + record.content)
```

Comment:
Check this gist, it's got the modified version of the ruby code that catches a lot of the exceptions, but there's two remaining.

The two offending keys are
2015-03-03 18:39:56.397 [INFO ] Parameters: {"name"=>"api", "type"=>"ssh-rsa", "content"=>"XYZ123=567[dfhhfl]"} (pid:80)